ada/language/ast.py

```diff
@@ -6736,8 +6736,15 @@ def parent_name_equation(typ=T.BaseTypeDecl.entity, root=T.Name):
                 lambda _: Bind(Self.type_var, No(T.AdaNode.entity).node),
             ) & Self.parent_name(root).as_entity.then(
                 lambda pn: pn.parent_name_equation(
-                    typ.comp_type(
-                        is_subscript=Not(Self.is_a(T.ExplicitDeref))
+                    # An ExplicitDeref will access the comp_type itself, so
```

Comment:
This comment seems to apply to the `typ.comp_type` below, not to the bit of special logic you added for access to subps. please:

1. Move the comment in the correct location
2. Add a comment about the new bit of logic